### DISCUSSION TITLE
after_create_account hook not working

### DISCUSSION THREAD

**Discussion by gczh** [janko/rodauth-rails#222] [Q&A] [answered]
I have the following code, following the documentation, in order to try and create a profile after creating an account via Google login using `google-oauth2` and `rodauth-omniauth`

Ideally, what I'd like to do is to create a Profile using the oauth data for First Name or Name field(s).

```ruby
# ==> Hooks
    # Validate custom fields in the create account form.
    # before_create_account do
    #   throw_error_status(422, "name", "must be present") if param("name").empty?
    # end

    omniauth_identity_insert_hash do
      super().merge({
        info: omniauth_info.to_json,
        credentials: omniauth_credentials.to_json,
        extra: omniauth_extra.to_json,
      })
    end

    # Perform additional actions after the account is created.
    after_create_account do
      # binding.pry
      Profile.create!(account_id: account_id)
    end
```

However, the `after_create_account` hook never seems to run any code within its block. Even the `binding.pry` doesn't run.

I even tried the following in my Account model

```ruby
class Account < ApplicationRecord
  include Rodauth::Rails.model

  enum :status, unverified: 1, verified: 2, closed: 3

  has_one :profile

  # after_create :create_profile
  after_create_commit :create_profile

  private

  def create_profile
    # binding.pry
    Profile.create!(account_id: account_id)
  end
end
```

The `binding.pry` doesn't run either. It seems like Accounts are being inserted directly into the database, thus bypassing even the rails model callbacks?

Wondering how I could figure out what's causing this behaviour and how to get around it.

**Comment by gczh** [comment, marked as answer]
Ok, I think I have this figured out after going through the `rodauth-omniauth` docs again.

I think the callbacks here are ignored entirely and deferred to `rodauth-omniauth`'s

Instead, I should do:

```ruby
after_omniauth_create_account do
  Profile.create(account_id: account_id, bio: omniauth_info["description"], image_url: omniauth_info["image"])
end
```

**Reply by janko** [reply]
Yes, the rationale is that account creation through OmniAuth login is an entirely different action than creating an account through the registration form (it uses separate endpoints with different params), so it deserves separate hooks.
